pkg/functions/openfaas/driver.go

```diff
@@ -94,6 +132,22 @@ func (d *ofDriver) Create(f *functions.Function, exec *functions.Exec) error {
 		}
 		return errors.Wrapf(err, "Error performing POST request, status: %v", res.StatusCode)
 	}
+
+	// make sure the function has started
+	return utils.Backoff(time.Duration(d.createTimeout)*time.Second, func() error {
```

Comment:
I'm thinking whether it would be better to use kubernetes informer and just listen for updates on that deployment (with a timeout if there no event that sets available replicas), but I've never used informer with deployments so... maybe some day ;)

Reply:
I've used informer on custom resources... it works, but it's also a bit of a pain.  I think this is fine.  We can optimize later if we really really need to.  But openfaas should also fix this.